src/main/java/org/broadinstitute/hellbender/tools/exome/AllelicCount.java

```diff
@@ -49,9 +79,61 @@ public AllelicCount(final SimpleInterval interval, final int refReadCount, final
 
     public int getAltReadCount() { return altReadCount; }
 
-    public Nucleotide getRefNucleotide() { return refNucleotide; }
+    public Nucleotide getRefNucleotide() {
+        return Utils.nonNull(refNucleotide, "The Ref nucleotide information is not available.");
+    }
+
+    public Nucleotide getAltNucleotide() {
+        return Utils.nonNull(altNucleotide, "The Alt nucleotide information is not available.");
+    }
+
+    public Integer getReadDepth() {
+        return Utils.nonNull(readDepth, "The read depth information is not available.");
+    }
+
+    public Double getHetLogOdds() {
+        return Utils.nonNull(hetLogOdds, "The log odds of heterozygosity is not available.");
+    }
 
-    public Nucleotide getAltNucleotide() { return altNucleotide; }
+    public static final Function<DataLine, AllelicCount> basicParser = dataLine -> {
```

Comment:
All these 3 lambdas could be declared as private methods (e.g. ```basicParser(...)```) and then make reference to them later (e.g. ```ClassName::basicParser```). 

Although they are effectively equivalent in this case, as matter of principle is better to postpone the added complexity of the lambda until it is really necessary.